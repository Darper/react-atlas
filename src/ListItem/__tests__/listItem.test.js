import React from "react";
import { mount } from "enzyme";
import { ListItem } from "../index";
import renderer from "react-test-renderer";

describe("Test ListGroup component", () => {
  it("Test render correctly", () => {
    const comp = <ListItem>Item one</ListItem>;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders children", () => {
    const li = mount(<ListItem>Item one</ListItem>);
    expect(li.text()).toBe("Item one");
  });

  it("renders className", () => {
    const li = mount(<ListItem className="myClass">Item one</ListItem>);
    expect(
      li
        .children()
        .first()
        .html()
    ).toContain("myClass");
    expect(li.find(".myClass").exists()).toBe(true);
  });

  it("sets id attribute", () => {
    const li = mount(
      <ListItem className="myClass" id="itemOne">
        Item one
      </ListItem>
    );

    expect(li.find('[id="itemOne"]').exists()).toBe(true);
  });

  it("renders leftItem", () => {
    const leftItem = `<div id="myLeftItem">left</div>`;
    const li = mount(
      <ListItem className="myClass" id="itemOne" leftItem={leftItem}>
        Item one
      </ListItem>
    );

    expect(li.html()).toContain("myLeftItem");
  });

  it("handles click", () => {
    const handleClick = jest.fn();
    const li = mount(
      <ListItem className="myClass" id="itemOne" onClick={handleClick}>
        Item one
      </ListItem>
    );

    li.simulate("click");
    expect(handleClick).toBeCalled();
  });

  it("renders rightItem", () => {
    const rightItem = `<div id="myRightItem">left</div>`;
    const li = mount(
      <ListItem className="myClass" id="itemOne" rightItem={rightItem}>
        Item one
      </ListItem>
    );

    expect(li.html()).toContain("myRightItem");
  });

  it("sets inline style", () => {
    const li = mount(
      <ListItem className="myClass" id="itemOne" style={{ "color": "red" }}>
        Item one
      </ListItem>
    );

    expect(
      li
        .children()
        .first()
        .html()
    ).toContain('style="color: red;"');
  });
});
