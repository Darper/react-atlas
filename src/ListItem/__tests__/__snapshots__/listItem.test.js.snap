// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ListGroup component Test render correctly 1`] = `
<div
  className="listItem"
  id={undefined}
  onClick={undefined}
  style={undefined}
>
  <div
    className="inlineBlock"
  >
    Item one
  </div>
</div>
`;
